javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll

```diff
@@ -191,4 +194,37 @@ module UnsafeShellCommandConstruction {
       )
     }
   }
+
+  private import semmle.javascript.dataflow.internal.AccessPaths
+  private import semmle.javascript.dataflow.InferredTypes
+
+  /**
+   * Holds if `instance` is an instance of the access-path `ap`, and there exists a guard
+   * that ensures that `instance` is not equal to `char`.
+   */
+  private predicate blocksCharInAccess(AccessPath ap, string char, Expr instance) {
```

Comment:
If the performance works out, we should try to move this `SanitizedChar` and `blocksCharInAccess` to `TaintTracking.qll` (lets make a separate issue for that).

Reply:
So generalize the sanitizer such that other queries can use it?